File: src/main/java/com/elchologamer/userlogin/listener/restriction/ItemPickupRestriction.java
package com.elchologamer.userlogin.listener.restriction;

import org.bukkit.event.EventHandler;
import org.bukkit.event.entity.EntityPickupItemEvent;

public class ItemPickupRestriction extends BaseRestriction<EntityPickupItemEvent> {

    public ItemPickupRestriction() {
        super("itemPickup");
    }

    @EventHandler
    public void handle(EntityPickupItemEvent e) {
        if (shouldRestrict(e))
            e.setCancelled(true);
    }
}

File: src/main/kotlin/com/elchologamer/userlogin/listener/restriction/ItemPickupRestriction.kt
package com.elchologamer.userlogin.listener.restriction

import org.bukkit.event.EventHandler
import org.bukkit.event.entity.EntityPickupItemEvent

object ItemPickupRestriction : BaseRestriction<EntityPickupItemEvent>("itemPickup") {
    @EventHandler
    fun handle(e: EntityPickupItemEvent) {
        if (shouldRestrict(e)) e.isCancelled = true
    }
}